src/wrapper_array.jl

```diff
@@ -168,6 +168,15 @@ for f in (:getindex, :view, :dotview)
         @propagate_inbounds function Base.$f(a::NamedDimsArray, inds...)
             # Some nonscalar case, will return an array, so need to give that names.
             data = Base.$f(parent(a), inds...)
+            if ndims(a) > 1 && inds isa Union{
+                    # Cases like nda[nda .> 0]
+                    Tuple{BitArray},
+                    Tuple{NamedDimsArray{L,T,N,<:BitArray}} where {L,T,N},
+                    # Cases like nda[findall(f, nda)]
+                    Tuple{AbstractVector{CartesianIndex{ndims(a)}}},
+                    }
```

Comment:
I don't like this because it isn't extensible, by other packages

Reply:
I don't love it but... what would a better design look like?

I guess you could make `remaining_dimnames_from_indexing` return `nothing` to signal return without names?

Reply:
Re-done but not sure it's right yet. 

Here's another error, which could also be caught by `remaining_dimnames_from_indexing`:
```julia
ndv = NamedDimsArray(rand(1:9, 10), :x)
ndv[rand(1:10, 2,3)] # ArgumentError: A 2 dimensional array, needs a 2-tuple of dimension names. Got: (:x,)
```